servicetalk-http-api/src/main/java/io/servicetalk/http/api/NetUtils.java

```diff
@@ -48,8 +52,8 @@ static boolean isValidIpV4Address(final CharSequence ip) {
 
     private static boolean isValidIpV4Address(final CharSequence ip, int from, int toExclusive) {
         return ip instanceof String ? isValidIpV4Address((String) ip, from, toExclusive, String::indexOf) :
-               ip instanceof AsciiBuffer ?
-                        isValidIpV4Address((AsciiBuffer) ip, from, toExclusive, AsciiBuffer::indexOf) :
+               isAsciiString(ip) ?
+                        isValidIpV4Address(ip, from, toExclusive, CharSequences::indexOf) :
```

Comment:
The `CharSequences::indexOf` will add unnecessary `instanceof` check for each character that we tried to avoid here using `AsciiBuffer::indexOf`. Looks like we need something like `CharSequences::asciiInfexOf` which will delegate to `AsciiBuffer::indexOf` without additional checks.

Reply:
I considered that possibility and convinced myself that this would be in-lined and the check pushed out of the loop. But too many 'woulds' in my sentence. I will put it to the test to confirm. Regardless, I think we can consider improving the `isValidIpV4Address` side rather than introducing an another `CharSequences` util. WDYT?

Reply:
If you have an idea for improvement, sure! Just trying to avoid any behavior changes in this PR. IIUC this code is used only when domain-specific cookies are present. We just don't have any test to catch the perf drop